src/spotless/groovy/net/neoforged/gradleutils/spotless/SpotlessUtilsExtension.groovy

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.gradleutils.spotless
+
+import com.diffplug.gradle.spotless.JavaExtension
+import com.diffplug.gradle.spotless.SpotlessExtension
+import com.diffplug.gradle.spotless.SpotlessTask
+import groovy.transform.CompileStatic
+import org.gradle.api.Project
+
+import javax.inject.Inject
+import java.nio.file.Files
+
+@CompileStatic
+abstract class SpotlessUtilsExtension {
+    final File configPath
+
+    @Inject
+    SpotlessUtilsExtension(Project project) {
+        configPath = new File(project.rootDir, '.gradle/formatter-config.xml')
+
+        final extract = project.tasks.register('extraGUSpotlessConfiguration', ExtractSpotlessConfiguration) {
+            it.output.set(configPath)
+        }
+
+        project.tasks.withType(SpotlessTask).configureEach {
+            it.dependsOn(extract)
+        }
+    }
+
+    void configure(SpotlessExtension ext) {
+        ext.java(this.&configure)
+    }
+
+    void configure(JavaExtension ext) {
+        // Spotless requires the config file to exist during configuration
+        if (!configPath.exists()) {
+            Files.createDirectories(configPath.toPath().parent)
+
+            try (
+                    final input = SpotlessUtilsExtension.getResourceAsStream('/formatter-config.xml');
+                    final output = Files.newOutputStream(configPath.toPath())
+            ) {
+                input.transferTo(output)
+            }
+        }
+
+        ext.endWithNewline()
```

Comment:
We should add the custom rule for "no star imports" as well.

Reply:
I am iffy on this, star imports can be especially useful in projects that use libraries like swing.

Reply:
Doesn't the IDE auto-import for you anyway? I'd rather standardize this than have to remember to add the rule to every project.

Reply:
Is it possible to disable such a custom rule from the project buildscript when applied by this extension?

Also, don't custom rules have a semi-weird version property which is used by Spotless to manually detect if the custom rules changed (as it is otherwise impossible to tell if a custom rule changed without checking bytecode)? We'd have to work on ensuring extension-defined custom rules and buildscript-defined custom rules can co-exist, and that we can change one or the other with the version property incrementing properly.

Reply:
I think, to avoid further delay on this PR, we will move consideration of adding the custom rule for banning wildcard imports to the future, particularly in light of my question on its potential implementation and effects.